llvm/lib/CodeGen/GlobalISel/GISelKnownBits.cpp

```diff
@@ -64,8 +64,11 @@ KnownBits GISelKnownBits::getKnownBits(MachineInstr &MI) {
 
 KnownBits GISelKnownBits::getKnownBits(Register R) {
   const LLT Ty = MRI.getType(R);
+  // Since the number of lanes in a scalable vector is unknown at compile time,
+  // we track one bit which is implicitly broadcast to all lanes.  This means
+  // that all lanes in a scalable vector are considered demanded.
   APInt DemandedElts =
-      Ty.isVector() ? APInt::getAllOnes(Ty.getNumElements()) : APInt(1, 1);
+      Ty.isFixedVector() ? APInt::getAllOnes(Ty.getNumElements()) : APInt(1, 1);
```

Comment:
Should this be a separate patch? What's the impact?

Reply:
There is a G_SPLAT_VECTOR test and a second combine zext_trunc_fold_matchinfo that relies on known bits to prove that zext(trunc(x)) is a noop. It crashed.

Reply:
The changes a copy-pasted from the DAG to support scalable vectors in known bits.